### PR TITLE
Exclude empty pages from search query for dashboard

### PR DESCRIPTION
**Pull request by JoeyStk**
### Short description

This PR changes how the number of outdated pages on the dashboard are counted. We want to take empty pages out (as they usually are only the parent/overview page to link to the children pages), but include empty pages when they are mirroring live content from another page, as the municipalities would recognize them as being "not empty".

### Proposed changes

- Add new filter to exclude empty pages (but include empty pages with mirrored content)
- Adapt the calculation of outdated pages on the dashboard
- Add tests for the dashboard

### Side effects

- I'm very unsure about the test test_number_of_outdated_pages_is_correct where I'm just checking if the string "Insgesamt X" is somewhere in the response. As there are a minimum of five rows this seems less than ideal, but was the best I could come up with. If you have any suggestion how a more stable test could look like, I'd be very happy to hear!
- I'm also not 100% sure about test_permission_to_view_chat. I wasn't sure what the most sensible way would be to test if the chat widget was there. Just checking if the title exists, seemed a bit too flaky for me and therefore I turned to a solution we used in the Compass, but maybe this is not appropriate here and then I will revert the changes.
- Ah, one thing I forget. I tested the new filter for ~400 pages and it took in my test runs between 1s and 2s. It was the slowest filter of them all, but they were still close and the Django bar said it was still only one SQL query, therefore I thought it might be okay and left it for now. If you have any suggestions on how to improve the speed of the filter, I'd be interested in hearing them too :) 

### Resolved issues

Fixes: #2660

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Review by PeterNerlich — CHANGES_REQUESTED** [review]
The functionality works well. I have few suggestions for the tests, I think it makes `test_number_of_outdated_pages_is_correct` a bit more robust. Only for `test_permission_to_view_chat` I'm not quite sure – I think we should check for the `form#chat-form` (maybe also that the `action` points to the correct endpoint) and the `textarea#id_text` and I was about to suggest a solution using regex, but it might be confusing and not as robust as I'd like, since I'd want to skip over HTML elements but rely on excluding `<` and `>` in wildcard matches in order to not skip over half the document to completely unrelated elements matching the second half by sheer chance.
I think the filters could get a general overhaul, and I think improving the performance of the empty pages filter would best fit in there as well. For now it's consistent with how the other filters are done and ok this way.

File: tests/cms/views/dashboard/test_dashboard.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+        )
+        .order_by("last_updated")
+        .exclude(Q(content="") & Q(page__mirrored_page=None))
+        .count()
+    )
+
+    assert response.status_code == 200
+    assert f"(Insgesamt {amount_of_outdated_pages})" in response.content.decode(
+        "utf-8"
+    )  # this is not ideal, it's a source for false positives.
```
[the remark this answers is not in the document]
Comment (JoeyStk) [reply tests/cms/views/dashboard/test_dashboard.py]: As noted and as we had discussed, I was also not happy with the test myself. I think this improves the test by a lot. I'd be okay with implementing it this way!

**Review by PeterNerlich — COMMENTED** [review]
Oh, something I didn't notice before, the button you check with `go-to-most-outdated-button.html` has `href="/augsburg/pages/de/1/edit/"` *(`1` is the page id, right?)*, but in `assert_button_leads_to_valid_page` you retrieve the URL for `"edit_page"` with `"page_id": 28`. I'm not sure whether I'm just overlooking something, but maybe it would also make sense to rewrite this as a regex match, which can extract any URL the button might link to and dynamically check that.

**Comment by JoeyStk** [comment]
The id has been corrected and changes should be up-to-date now :) Peter and I decided that for now we won't rewrite this is a regex

**Review by PeterNerlich — APPROVED** [review]
Looks good!

**Comment by MizukiTemma** [comment]
Looks good and working as expected so far as I tested 👍 

Only one question: in the issue it was suggested that only published pages should be shown. Was this idea abandoned?

**Comment by JoeyStk** [comment]
> Only one question: in the issue it was suggested that only published pages should be shown. Was this idea abandoned?

This question took some time to be answered. Today it was decided that drafts should not be shown in the "<outdated pages" section, but instead get a section for themselves, see #2765. 

I will adapt the change now :)

**Comment by JoeyStk** [comment]
@MizukiTemma / @david-venhoff PR is ready for review again :)

**Review by MizukiTemma — APPROVED** [review]
Thank you!

**Pull request merged**